staging/src/k8s.io/kubectl/pkg/cmd/proxy/proxy.go

```diff
@@ -185,6 +185,13 @@ func (o *ProxyOptions) Complete(f cmdutil.Factory) error {
 			RejectMethods: proxy.MakeRegexpArrayOrDie(o.rejectMethods),
 		}
 	}
+	if o.address == "0.0.0.0" {
+		klog.Warning("kubectl proxy will serve on all network interfaces (0.0.0.0), which may expose cluster access via the proxy to anyone who can access your network.")
+	}
+	// string contains will use exact match, not regex.
+	if strings.Contains(o.acceptHosts, ".*") {
+		klog.Warning("--accept-hosts will accept any host, which may expose cluster access via the proxy to anyone who can access your network.")
```

Comment:
this warning isn't necessarily true. `.*\.example.com` contains `.*` but does not accept any host

Reply:
I think `.*\.example.com` will still accept kind of any.  The range is too wide. But we can modify the warning sentence to be `--accept-hosts will accept a wide range of hosts, which may expose cluster access via the proxy to anyone who can access your network.` What do you think

Reply:
> I think .*\.example.com will still accept kind of any.

I don't think it does? It only accepts subdomains of example.com, right?

> The range is too wide

Not if you're intending to allow all subdomains of example.com, right?

Reply:
I think there's two heuristics we can apply here (we could do both):

1. actually attempt to match a random host string we construct ("$(random-uuid).$(random-uuid)" or something) against the acceptHosts pattern... if it gets accepted that's a good sign *any* host would be accepted
2. detect `.*` and warn generically about a broad accept-hosts setting as you indicated in https://github.com/kubernetes/kubernetes/pull/122816#discussion_r1473168623

Reply:
> > I think .*.example.com will still accept kind of any.
> 
> I don't think it does? It only accepts subdomains of example.com, right?
> 
> > The range is too wide
> 
> Not if you're intending to allow all subdomains of example.com, right?

Discussed with Michael, we think the sub domain is still very broad. What about we do both check. random match pattern OR contains `.*`

Reply:
Warning for allowing any subdomain is too broad imo. We should stick to warnings for allowing the world and not try to get in the weeds of what constitutes "appropriately restricted" because this will be different for every organization.